website/src/app/users/users.component.ts

```diff
@@ -0,0 +1,25 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import {User} from '../user';
+
+@Component({
+  selector: 'app-users',
+  templateUrl: './users.component.html',
+  styleUrls: ['./users.component.scss']
+})
+export class UsersComponent {
+
+  constructor() { }
+
+  ngOnInit() {
+  }
+
+  @Input() 
+  users: User[];
+
+  @Output()
+  remove: EventEmitter<User> = new EventEmitter();
+
+  onRemoveUser(user: User) {
+    this.remove.emit(user);
```

Comment:
Since all the logic is in AppComponent, you basically handle event from Useritem here and emit it further up the tree. Why it couldn't have been handled here or even part of it in Useritem? Architecture-wise you want every component to perform actions (by delegating to relevant service) and only pass events when the parent components need to react to those events. I.e. Useritem handles deletion, but lets Users know, so Users can refresh the list. AppComponent doesn't have to be involved.